byterun/lockfree.c

```diff
@@ -0,0 +1,60 @@
+#define CAML_INTERNALS
```

Comment:
This c file is generating a bunch of warnings about type conversion. Are we able to check them out and fix them up if they are ok?